instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpHeaderAttributes.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.http;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.instrumentation.api.caching.Cache;
+import java.util.List;
+
+final class HttpHeaderAttributes {
+
+  private static final Cache<String, AttributeKey<List<String>>> requestKeysCache =
+      Cache.newBuilder().setMaximumSize(32).build();
+  private static final Cache<String, AttributeKey<List<String>>> responseKeysCache =
+      Cache.newBuilder().setMaximumSize(32).build();
+
```

Comment:
We'd create different `AttributeKey` values for same headers, if there were more than 1 instrumentations loaded.
And I wanted to avoid exposing the attribute keys in any public APIs (although now that I think of it, it could be implemented without doing so)